emcc.py

```diff
@@ -1295,18 +1303,22 @@ def phase_setup(options, state, newargs, settings_map):
   state.has_dash_c = '-c' in newargs
   state.has_dash_S = '-S' in newargs
   state.has_dash_E = '-E' in newargs
-  state.preprocess_only = state.has_dash_E or '-M' in newargs or '-MM' in newargs or '-fsyntax-only' in newargs
-  state.compile_only = state.has_dash_c or state.has_dash_S or state.has_header_inputs or state.preprocess_only
 
-  if state.compile_only:
+  if has_header_inputs:
+    state.mode = Mode.PCH
+  if options.post_link:
```

Comment:
is this intended to be `elif` or are the latter options just supposed to override PCH mode?

Reply:
Good question..  

I've updated this block to reflect the prior logic, which I believe is also what clang and gcc both do.

1. First check for `-E` or other flags in which case just pre-process.
2. Next check for any `.h` files as input, in which case use PCH mode.
3. Next look for `-c` in which case use compile-only mode
4. Otherwise do the default compile + link